### PR TITLE
Fixes #20: Add parameter for github access token for an increased rate limit

### LINKED ISSUE #20: Add parameter for github access token
Unauthenticated requests without github access tokens are **rate limited** to 60 requests per hour, and the application may send many requests in one run. A parameter should be added to allow users to add their access token to the requests.

```bash
-t, --token <github access token> (optional)
  A github access token to be used in sending requests. generate one from here (https://github.com/settings/tokens?type=beta)

# usage example
$ starred_search -u 'link-' -f 'es6' -t github_pat_*****
```

### PR DESCRIPTION
**Pull request by Raamyy**
Fixes #20

Add parameter for github access token for an increased rate limit

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ Options:
 
   -d, --debug
     Outputs stack trace in case an exception is thrown
+  
+  -t, --token
+  A github access token to be used in sending requests. generate one from here (https://github.com/settings/tokens?type=beta)
```
Comment (Link-) [inline src/cli.js]: Reading a token as plain text argument is not secure. It'll expose the token to the shell history and could lead to bigger problems.

How about we change this to read the token from an environment variable such as `$GITHUB_TOKEN`?

**Review by Link- — CHANGES_REQUESTED** [review]
I like this addition, thank you 🙏 let's just make it more secure

File: src/cli.js [hunk 2]
```diff
@@ -4,12 +4,16 @@ const main = require('./main.js');
...
 const cachedir = require('cachedir');
+require('dotenv').config()
 
 const help = `
 Usage: starred_search [OPTIONS] [ARGS]...
 
   Search your or any other user's starred repositories on GitHub for a keyword.
 
+  Reads from $GITHUB_TOKEN env variable an oprtional github access token to be used in sending requests.
+  Generate one from here https://github.com/settings/tokens?type=beta
```
Comment (Link-) [inline src/cli.js]:
```suggestion
```

Let's skip it here, we'll add it to the README

**Comment by Raamyy** [comment]
@Link- I have resolved it, and I added `dotenv` package so it can be read from .env file also. However this resulted in making bunch of changes in the `package.lock` file. I think is because I'm using a bit older version of `npm`. So is it okay having it with these changes or should I do something specific 🤔

File: src/cli.js (hunk 2 above)
Comment (Raamyy) [reply src/cli.js]: Okay removed from the `-h` arg, how you want it shown in the readme?

**Review by Link- — APPROVED** [review]
Let's 🚢  it

**Pull request merged**